### PR TITLE
feat: Add api to remove a community private key (nim part)

### LINKED ISSUE #11475: Add API to forget/remove the community private key (ownership transfer)
### Description

Add an API in status-go and Nim functions that call that API so that an owner can forget/remove the private key from their DB.

This is part of the flow where an owner would transfer ownership to someone else or to another PC. This is to make sure there is never 2 control nodes at the same time.

Just before deleting the private key, the node should update the description and remove itself as owner and send it to members. Then it deletes the key.

When another node imports the community private key, the invert happens, it imports, then updates the description, marking itself as owner (it might be already happening, but double-check).

There is one edge case where you import the key in the same account on another machine, then you have two control nodes. After that, you remove the key from one of the machine, removing yourself as owner. Then for a short while, the community would appear to have no owners. Either make it that the new control node corrects the record straight away by sending a new description with it as the owner or when the next periodic community description is published, make sure the record is set straight (that account is still the owner).

### Acceptance Criteria

- There is a status-go function and API to remove the private key, plus tests
  - by doing so, the node updates the community description and removed itself
- The new control node advertises itself as the control node
- There is Nim code to call that status-go API from the QML

### PR DESCRIPTION
**Pull request by MishkaRogachev**
Close #11475

### What does the PR do

Add endpoint to remove community private key (owner node)

### Affected areas

Communities

### CODE REVIEW

File: src/app/modules/main/communities/view.nim [hunk 1]
```diff
@@ -495,6 +495,9 @@ QtObject:
   proc isCommunityRequestPending*(self: View, communityId: string): bool {.slot.} =
     self.delegate.isCommunityRequestPending(communityId)
 
+  proc removePrivateKey*(self: View, communityKey: string) {.slot.} =
```
Comment (alexjba) [inline src/app/modules/main/communities/view.nim:498]:
```suggestion
  proc removePrivateKey*(self: View, communityId: string) {.slot.} =
```

Comment (alexjba) [reply src/app/modules/main/communities/view.nim:498]: Shouldn't be `communityId`? Same for all proc

**Review by alexjba — COMMENTED** [review]
LGMT! Tested and the private key is removed by using this API. 👍 Nice work!

One thing I've noticed though is that the `isControlNode` community property is not updated after the control node is removed. Is it in the scope of this task, or we need another one?

File: src/app/modules/main/communities/view.nim (hunk 1 above)
Comment (MishkaRogachev) [reply src/app/modules/main/communities/view.nim]: Fixed!

**Review by jrainville — APPROVED** [review]

**Comment by jrainville** [comment]
> One thing I've noticed though is that the isControlNode community property is not updated after the control node is removed. Is it in the scope of this task, or we need another one?

It should be in the scope for this task. Can you fix please @MishkaRogachev

**Review by osmaczko — APPROVED** [review]

**Comment by MishkaRogachev** [comment]
> > One thing I've noticed though is that the isControlNode community property is not updated after the control node is removed. Is it in the scope of this task, or we need another one?
> 
> It should be in the scope for this task. Can you fix please @MishkaRogachev

That should be fixed currently

**Comment by alexjba** [comment]
> > > One thing I've noticed though is that the isControlNode community property is not updated after the control node is removed. Is it in the scope of this task, or we need another one?
> > 
> > 
> > It should be in the scope for this task. Can you fix please @MishkaRogachev
> 
> That should be fixed currently

Not sure why the event doesn't propagate to the `sectionItemModel` used in QML. Do you think we're missing a signal from nim?

**Comment by alexjba** [comment]
Importing the private key works as expected. The community is properly updated.

**Review by alexjba — APPROVED** [review]
@MishkaRogachev Thanks for fixing that!

**Pull request merged**